kornia/feature/sold2/sold2_detector.py

```diff
@@ -0,0 +1,763 @@
+             - candidate_junc_end.to(torch.float32)) ** 2, dim=-1))
+        normalized_seg_length = (segments_length
+                                 / (((H ** 2) + (W ** 2)) ** 0.5))
+
+        # Perform local max search
+        num_cand = cand_h.shape[0]
+        group_size = 10000
+        if num_cand > group_size:
+            num_iter = math.ceil(num_cand / group_size)
+            sampled_feat_lst = []
+            for iter_idx in range(num_iter):
+                if not iter_idx == num_iter - 1:
+                    cand_h_ = cand_h[iter_idx * group_size:
+                                     (iter_idx + 1) * group_size, :]
+                    cand_w_ = cand_w[iter_idx * group_size:
+                                     (iter_idx + 1) * group_size, :]
+                    normalized_seg_length_ = normalized_seg_length[
+                        iter_idx * group_size: (iter_idx + 1) * group_size]
+                else:
+                    cand_h_ = cand_h[iter_idx * group_size:, :]
+                    cand_w_ = cand_w[iter_idx * group_size:, :]
+                    normalized_seg_length_ = normalized_seg_length[
+                        iter_idx * group_size:]
+                sampled_feat_ = self.detect_local_max(
+                    heatmap, cand_h_, cand_w_, H, W,
+                    normalized_seg_length_, device)
+                sampled_feat_lst.append(sampled_feat_)
+            sampled_feat = torch.cat(sampled_feat_lst, dim=0)
+        else:
+            sampled_feat = self.detect_local_max(
+                heatmap, cand_h, cand_w, H, W,
+                normalized_seg_length, device)
+
+        # [Simple threshold detection]
+        # detection_results is a mask over all candidates
+        detection_results = (torch.mean(sampled_feat, dim=-1)
+                             > self.detect_thresh)
+
+        # [Inlier threshold detection]
+        if self.inlier_thresh > 0.:
+            inlier_ratio = torch.sum(
+                sampled_feat > self.detect_thresh,
+                dim=-1).to(torch.float32) / self.num_samples
+            detection_results_inlier = inlier_ratio >= self.inlier_thresh
+            detection_results = detection_results * detection_results_inlier
+
+        # Convert detection results back to line_map_pred
+        detected_junc_indexes = candidate_index_map[detection_results, :]
+        line_map_pred[detected_junc_indexes[:, 0],
+                      detected_junc_indexes[:, 1]] = 1
+        line_map_pred[detected_junc_indexes[:, 1],
+                      detected_junc_indexes[:, 0]] = 1
+
+        # [Junction refinement]
+        if self.use_junction_refinement and len(detected_junc_indexes) > 0:
+            junctions, line_map_pred = self.refine_junction_perturb(
+                junctions, line_map_pred, heatmap, H, W, device)
+
+        return line_map_pred, junctions, heatmap
+
+    def refine_heatmap(self, heatmap, ratio=0.2, valid_thresh=1e-2):
```

Comment:
```suggestion
    def refine_heatmap(self, heatmap: Tensor, ratio: float = 0.2, valid_thresh: float = 1e-2) -> Tensor:
```
revisit if you have some time the typings